### ISSUE TITLE
codimension installation problem with ubuntu 19.04

### ISSUE THREAD

**Issue by thnd23** [SergeySatskiy/codimension#456]
There is currently a problem with an installation of codimension with ubuntu 19.04. By default, os installs Python 3.7 and using 3.6.8 did not lead to any useful results.

**Comment by SergeySatskiy** [comment]
Could you please post the output with error messages?

**Comment by thnd23** [comment]
Error message associated with failing python3.7 installation is: "Failed building wheel for cdmcfparser". This occurs due to compilation error associated with C: invalid conversion from 'const char*' to 'char*". Probably warnings are treated as errors? It also mentions DeprecatedWarning on convert() python calls. Is this application only intended to be used with python3.6? I have noticed such a remark at pypi. I have noticed that installation with 3.6 did not happen since it was still used as an incorrect symbolic link to 3.7 (while compiling in virtualenv that used 3.6). Would the entire output be helpful?

**Comment by SergeySatskiy** [comment]
Thank you, I think it is enough for now.

It seems that the problem is with the cdmcfparser compilation - it is a python extension module written in C/C++. 19.04 uses a newer GCC so it seems the compiler is stricter than the older one.

Codimension should work on python 3.7, I don't see major changes in the language grammar (and in the produced AST respectively) so at most I expect minor changes.
However to give it a try and fix the problems I would have to install 19.04. I cannot promise to fix everything quick. Optimistically (if no major changes are required) I would be able to prepare a new release by the end of the weekend.

**Comment by thnd23** [comment]
Thank you for almost instant response, I will definitely look at it at the end of the weekend and try to set the tooling up with python3.6.

**Comment by SergeySatskiy** [comment]
I had some time yesterday and found what caused the problem. This is a compilation of the third party library called pycxx. I used the version 7.0.1 and the problem is fixed in 7.1.2. The update of the library helped.

On top of it python 3.7 changed its grammar. Now the 'async' and 'await' are proper keywords while in python 3.6 they were tokens. I managed to make the appropriate changes.

Some more testing is still required plus some changes would need to be done in another extension module. Then re-packaging of two modules plus the IDE. The new version will support 3.7 and will be installable on 19.04
